### PR TITLE
RavenDB-17745 fix test

### PR DESCRIPTION
**Pull request by efratshenhar**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-17745

### Additional description

The test was passing always. 

_Please delete below the options that are not relevant_

### Type of change

- Test fix

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing by Contributor

- Tests have been added that prove the fix is effective or that the feature works

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs [hunk 1]
```diff
@@ -299,13 +299,20 @@ private class TimerState
 
         private static void HandleHeartbeat(object state)
         {
-            var timerState = (TimerState)state;
-            if (timerState.Parent.TryGetTarget(out BulkInsertOperation bulkInsert) == false)
+            try
```
Comment (ppekrol) [inline src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs]: the only part here that might fail is the dispose of the timer maybe? wrap it up there?

File: src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs [hunk 2]
```diff
@@ -299,13 +299,20 @@ private class TimerState
...
             {
-                timerState.Timer.Dispose();
-                return;
+                var timerState = (TimerState)state;
+                if (timerState.Parent.TryGetTarget(out BulkInsertOperation bulkInsert) == false)
+                {
+                    timerState.Timer.Dispose();
+                    return;
+                }
+                _ = bulkInsert.SendHeartBeatAsync();
```
Comment (ppekrol) [inline src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs]: you need to add try/catch inside the SendHeartBeatAsync as well, this is a background task, not awaited so wrapping it in try/catch does not yield any results

File: src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs [hunk 3]
```diff
@@ -299,13 +299,20 @@ private class TimerState
...
+            }
+            catch (Exception e)
```
Comment (ppekrol) [inline src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs]: warning here

File: src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs (hunk 1 above)
Comment (efratshenhar) [reply src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs]: Done

File: src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs (hunk 3 above)
Comment (efratshenhar) [reply src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs]: Done

File: src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs (hunk 2 above)
Comment (efratshenhar) [reply src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs]: Done

**Pull request merged**